### PR TITLE
Add patch-antimalware-scan-interface-function.yml and updated patch-e…

### PR DESCRIPTION
**Pull request by jtothej**
Add patch-antimalware-scan-interface-function.yml and updated patch-event-tracing-for-windows-function.yml

### CODE REVIEW

File: anti-analysis/anti-av/patch-antimalware-scan-interface-function.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      - match: link function at runtime on Windows
+      - or:
+        - api: kernel32.VirtualProtect
+        - api: kernel32.VirtualProtectEx
+        - api: ntdll.NtProtectVirtualMemory
+        - api: ZwProtectVirtualMemory
+        - string: "VirtualProtect"
+        - string: "VirtualProtectEx"
+        - string: "NtProtectVirtualMemory"
+        - string: "ZwProtectVirtualMemory"
```
Comment (williballenthin) [inline anti-analysis/anti-av/patch-antimalware-scan-interface-function.yml:27]: i think we should factor this logic out into a library rule, like `patch read-only memory at runtime` or something

Comment (jtothej) [reply anti-analysis/anti-av/patch-antimalware-scan-interface-function.yml:27]: I was initially thinking about using `allocate RWX memory` (which in turn depends on `allocate memory`) but `allocate memory` matches on both memory allocation functions and functions that change of protection of already allocated memory.
So maybe we could split `allocate memory` to actual `allocate memory` and something like `change memory protection` (including `VirtualProtect`, `VirtualProtectEx`, `NtProtectVirtualMemory`, `ZwProtectVirtualMemory`). The we could have `change memory protection to RWX` and use it in rules in this PR. The problem with this approach would be figuring out which rules currently using `allocate memory` rule depend on memory allocation functions, which depend on change of memory functions and which on both.

Comment (williballenthin) [reply anti-analysis/anti-av/patch-antimalware-scan-interface-function.yml:27]: yup, that sounds like a great refactor!

Comment (mr-tz) [reply anti-analysis/anti-av/patch-antimalware-scan-interface-function.yml:27]: refactoring this in #836

Comment (mr-tz) [inline anti-analysis/anti-av/patch-antimalware-scan-interface-function.yml:27]:
```suggestion
      - match: change memory protection
```

File: anti-analysis/anti-av/patch-event-tracing-for-windows-function.yml [hunk 2]
```diff
@@ -19,9 +19,11 @@ rule:
       - match: link function at runtime on Windows
       - or:
         - api: kernel32.VirtualProtect
+        - api: kernel32.VirtualProtectEx
         - api: ntdll.NtProtectVirtualMemory  # exported by only ntdll, not ntoskrnl
         - api: ZwProtectVirtualMemory  # exported by both ntdll and ntoskrnl
         - string: "VirtualProtect"
+        - string: "VirtualProtectEx"
         - string: "NtProtectVirtualMemory"
         - string: "ZwProtectVirtualMemory"
```
Comment (mr-tz) [inline anti-analysis/anti-av/patch-event-tracing-for-windows-function.yml:28]:
```suggestion
      - match: change memory protection
```

**Pull request merged**